core/consensus/pow/src/lib.rs

```diff
@@ -205,11 +226,12 @@ impl<B: BlockT<Hash=H256>, C, Algorithm> Verifier<B> for PowVerifier<C, Algorith
 		let best_aux = PowAux::read(self.client.as_ref(), &best_hash)?;
 		let mut aux = PowAux::read(self.client.as_ref(), &parent_hash)?;
 
-		let (checked_header, difficulty, seal) = self.check_header::<B>(
+		let (checked_header, difficulty, seal) = self.check_header(
 			header,
 			BlockId::Hash(parent_hash),
 		)?;
-		aux.total_difficulty = aux.total_difficulty.saturating_add(difficulty);
+		aux.difficulty = difficulty;
+		aux.total_difficulty.add(difficulty);
```

Comment:
Why was this changed from a `saturating_add()`?

Reply:
This is the `add` function from `TotalDifficulty` trait. The idea here is that some `Difficulty` definitions may want to use its own customized method for calculating the total difficulty from current difficulty. Most of the time `TotalDifficulty::add` will still be implemented as `saturating_add`.